### PR TITLE
XMC CAN Sending & Unity starting point 

### PR DESCRIPTION
**Pull request by LinjingZhang**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Enable all CAN basic function s

Related Issue
n/a

Context
https://jirard.intra.infineon.com/browse/DESMAKERS-3775 (CAN API)
https://jirard.intra.infineon.com/browse/DESMAKERS-3821 (Unity Test) - Sorry again I will try to separate the PR next time... 🐱 

The XMC CAN module works a bit different: It will not receive all data frames and then filter them. Instead, it only accept data frame with defined ID/ ID Mask. Spent two days reading the user manual and experimenting, no success

⚠️ Therefore the extra setDeviceId() function is added and I could not implement the filtering feature.

### CODE REVIEW

**Comment by LinjingZhang** [comment]
### Example program
make FQBN=Infineon:xmc:XMC4700_Relax_Kit PORT=COM20 CANReceiver flash monitor

make FQBN=Infineon:xmc:XMC4700_Relax_Kit PORT=COM20 CANReceiverCallback flash monitor

make FQBN=Infineon:xmc:XMC4700_Relax_Kit PORT=COM24 CANSender flash monitor

only example flash is enable, unit test in development

File: libraries/CAN/src/CANXMC.cpp [hunk 1]
```diff
@@ -33,9 +42,14 @@ namespace ifx
   }
   CANXMC::~CANXMC() {}
 
+  int CANXMC::setDeviceId(long id = 0x12) { // TODO: delete in the future!
```
Comment (jaenrig-ifx) [inline libraries/CAN/src/CANXMC.cpp:45]: This id should not have a default value I guess?

**Review by jaenrig-ifx — CHANGES_REQUESTED** [review]
Getting there! 👍 

Unity should not be part of the code sources... but specific configuration files for this project. The rest should be able to be pulled-in with some external dependency management system providing the required unity version.

File: libraries/CAN/src/CANXMC.cpp (hunk 1 above)
Comment (LinjingZhang) [reply libraries/CAN/src/CANXMC.cpp]: fix

**Review by jaenrig-ifx — APPROVED** [review]
Pending changes postponed. Future PRs.

**Pull request merged**